packages/contracts-bedrock/src/L1/winddown/BalanceClaimer.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+// Interfaces
+import { IBalanceWithdrawer } from "src/L1/interfaces/winddown/IBalanceWithdrawer.sol";
+import { IEthBalanceWithdrawer } from "src/L1/interfaces/winddown/IEthBalanceWithdrawer.sol";
+import { IErc20BalanceWithdrawer } from "src/L1/interfaces/winddown/IErc20BalanceWithdrawer.sol";
+import { IBalanceClaimer } from "src/L1/interfaces/winddown/IBalanceClaimer.sol";
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import { ISemver } from "src/universal/interfaces/ISemver.sol";
+
+/// @title BalanceClaimer
+/// @notice Contract that allows users to claim and withdraw their balances
+contract BalanceClaimer is Initializable, ISemver {
+    /// @notice Emitted when a user claims their balance
+    event BalanceClaimed(
+        address indexed user, uint256 ethBalance, IBalanceWithdrawer.Erc20BalanceClaim[] erc20TokenBalances
+    );
+
+    string public constant version = "1.0.0";
+
+    /// @notice the root of the merkle tree
+    bytes32 public root;
+
+    /// @notice OptimismPortal proxy address
+    IEthBalanceWithdrawer public ethBalanceWithdrawer;
+
+    /// @notice L1StandardBridge proxy address
+    IErc20BalanceWithdrawer public erc20BalanceWithdrawer;
```

Comment:
Why those vars aren't called `portal` and `l1StandardBridge`? Isn't this feature designed to interact specifically with those both contracts? In that case, I feel it'd more accurate.
But maybe you opted-out from this to make it more future proof?

Reply:
If it can be fixed to the l1 bridge and portal, we could even hardcode those values as constants from the predeploys

Reply:
agree. i'd rename them to the actual contract names

Reply:
+1

Reply:
because we are building something generic that can be applied to other projects that require claiming tokens or ETH, so it's not attached to L1StandardBridge or OptmismPortal.

Reply:
The BalanceClaimer contract/interface should not know what contract specifics is each withdrawer, because it's completely unrelated